File: src/main/java/io/playground/orderservice/application/saga/port/client/InventoryClientPort.java
package io.playground.orderservice.application.saga.port.client;

import io.playground.orderservice.application.saga.dto.ClientDto;

import java.util.List;

public interface InventoryClientPort {
    void reserveStocks(String orderExternalId,
                       List<ClientDto.ReservationRequest> infos);

    void confirmStocks(String orderExternalId);
}

File: src/main/kotlin/io/playground/orderservice/application/saga/port/client/InventoryClientPort.kt
package io.playground.orderservice.application.saga.port.client

import io.playground.orderservice.application.saga.dto.ClientDto

interface InventoryClientPort {
    fun reserveStocks(orderExternalId: String, infos: List<ClientDto.ReservationRequest>)

    fun confirmStocks(orderExternalId: String)
}
